app/controllers/users.js

```diff
@@ -445,3 +446,41 @@ exports.searchUser = (req, res) => {
       });
     });
 };
+
+
+exports.profile = (req, res) => {
+  const { id } = req.params;
+  const details = {};
+  User.findById({ _id: id }).then((user) => {
+    if (!user) {
+      return res.status(404).json({
+        message: 'User Not Found',
+      });
+    }
+    Game.find({ gameWinner: user.username })
+      .then((games) => {
+        details.id = user._id;
+        details.email = user.email;
+        details.name = user.name;
+        details.username = user.username;
+        details.image = user.profileImage;
+        details.gamesWon = games.length;
+        Game.find().exec((err, games) => {
+          if (err) {
+            return res.status(400).json({
+              message: 'Error Occured'
+            });
+          }
+          const userGameLog = games.map(game => ({
+            gameId: game.gameId,
+            playedAt: game.played,
+            log: game.players
+              .filter(player => player.username === user.username),
+            gameWinner: game.gameWinner === user.username ? 'WON' : 'LOST'
+          }));
+          details.userGame = userGameLog.filter(y => y.log.length !== 0);
```

Comment:
I'm finding it hard to understand the variable `y` is. Use a descriptive variable name. @Rhotimee